custom-checks/checkstyle/src/main/java/org/openhab/tools/analysis/utils/SatCheckUtils.java

```diff
@@ -0,0 +1,101 @@
+import java.util.Queue;
+
+import com.puppycrawl.tools.checkstyle.api.DetailAST;
+import com.puppycrawl.tools.checkstyle.utils.CheckUtils;
+
+/**
+ * Class that contains utility methods for checks.
+ *
+ * @author Kristina Simova - Initial contribution
+ * @author Tanya Georgieva - Added methods that are looking for children nodes of a given type
+ *
+ */
+public final class SatCheckUtils {
+
+    /**
+     * Creates a new FullIdent starting from the given node and returns the fully qualified name as a String.
+     *
+     * @param typeAST a type node
+     * @return the fully qualified name as a String
+     */
+    public static String createFullyQualifiedName(DetailAST typeAST) {
+        return CheckUtils.createFullType(typeAST).getText();
+    }
+
+    /**
+     * Method that returns the first (direct or indirect) child node of a given type or null if such does not exist
+     *
+     * @param ast the ast
+     * @param type the type of child node we are looking for
+     * @return the first child node of a given type or null if such does not exist
+     */
+    public static DetailAST getFirstNodeOfType(DetailAST ast, int type) {
+        if (ast == null) {
+            return null;
+        }
+        if (ast.getType() == type) {
+            return ast;
+        }
+        DetailAST nextNode = getFirstNodeOfType(ast.getFirstChild(), type);
+        DetailAST nextSibling = getFirstNodeOfType(ast.getNextSibling(), type);
+        // if the current node does not have a child node return its sibling
+        return nextNode != null ? nextNode : nextSibling;
+    }
+
+    /**
+     * Method that searches for all direct children nodes of a given type and returns them
+     *
+     * @param ast the ast
+     * @param type the type of children nodes we are looking for
+     * @return list of the ast's children of a given type
+     */
+    public static LinkedList<DetailAST> getAllChildrenNodesOfType(DetailAST ast, int type) {
+        LinkedList<DetailAST> childrenNodes = new LinkedList<>();
+        Queue<DetailAST> visitedNodes = new LinkedList<>();
+        visitedNodes.add(ast.getFirstChild());
+
+        while (!visitedNodes.isEmpty()) {
+            DetailAST currentNode = visitedNodes.poll();
+            if (currentNode.getType() == type) {
+                childrenNodes.add(currentNode);
+            }
+            DetailAST nodeNextSibling = currentNode.getNextSibling();
+            if (nodeNextSibling != null) {
+                visitedNodes.add(nodeNextSibling);
+            }
+        }
+        return childrenNodes;
+    }
+
+    /**
+     * Method that searches for all direct and indirect children nodes of a given type and returns them
+     *
+     * @param requiredNodes list of the required nodes
+     * @param ast the ast
+     * @param type the token type to match
+     * @return list of the required nodes
+     */
+    public static LinkedList<DetailAST> getAllNodesOfType(LinkedList<DetailAST> requiredNodes, DetailAST ast,
```

Comment:
Is it really necessary to expose the `LinkedList` implementation on all these methods? 
Using `List` on all these methods would make them easier to reuse.

Reply:
@wborn I will fix this when I update the PR. Thank you :)